Frends.Community.Pgp/Frends.Community.Pgp/PgpTasks.cs

```diff
@@ -0,0 +1,549 @@
+            if (enumerator.MoveNext())
+            {
+                spGen.SetSignerUserId(false, (string)enumerator.Current);
+                sGen.SetHashedSubpackets(spGen.Generate());
+            }
+
+            Stream fIn = File.OpenRead(input.InputFile);
+            Stream outputStream = File.Create(input.OutputFile);
+
+            ArmoredOutputStream aOut = new ArmoredOutputStream(outputStream);
+
+            aOut.BeginClearText(digest);
+
+            //
+            // note the last \n/\r/\r\n in the file is ignored
+            //
+            MemoryStream lineOut = new MemoryStream();
+            int lookAhead = Services.ReadInputLine(lineOut, fIn);
+
+            Services.ProcessLine(aOut, sGen, lineOut.ToArray());
+
+            if (lookAhead != -1)
+            {
+                do
+                {
+                    lookAhead = Services.ReadInputLine(lineOut, lookAhead, fIn);
+
+                    sGen.Update((byte)'\r');
+                    sGen.Update((byte)'\n');
+
+                    Services.ProcessLine(aOut, sGen, lineOut.ToArray());
+                }
+                while (lookAhead != -1);
+            }
+
+            fIn.Close();
+
+            aOut.EndClearText();
+
+            BcpgOutputStream bOut = new BcpgOutputStream(aOut);
+
+            sGen.Generate().Encode(bOut);
+
+            aOut.Close();
+            outputStream.Close();
+
+            PgpClearTextSignatureResult ret = new PgpClearTextSignatureResult
+            {
+                FilePath = input.OutputFile
+            };
+
+            return ret;
+        }
+
+        #endregion
+
+        #region PgpVerifyFileClearTextSignature 
+        /// <summary>
+        /// Verifies clear text PGP signature. See documentation at https://github.com/CommunityHiQ/Frends.Community.PgpVerifyClearTextSignature Returns: Object {string FilePath, Boolean Verified}
+        /// </summary>
+        public static PgpVerifyClearTextSignatureResult PgpVerifyFileClearTextSignature(PgpVerifyClearTextSignatureInput input)
+        {
+            using (var inStr = File.OpenRead(input.InputFile))
+            using (var outStr = File.Create(input.OutputFile))
+            using (var keyStr = PgpUtilities.GetDecoderStream(File.OpenRead(input.PublicKeyFile)))
+            {
+                ArmoredInputStream aInputStr = new ArmoredInputStream(inStr);
+
+                //
+                // write out signed section using the local line separator.
+                // note: trailing white space needs to be removed from the end of
+                // each line RFC 4880 Section 7.1
+                //
+                MemoryStream lineOut = new MemoryStream();
+                int lookAhead = Services.PgpVerifyClearTextSignatureReadInputLine(lineOut, aInputStr);
+                byte[] lineSep = Encoding.ASCII.GetBytes(Environment.NewLine); ;
+
+
+                if (lookAhead != -1 && aInputStr.IsClearText())
```

Comment:
This if+while+else seems like it could be a lot more simple with a guard clause for a single line file and then just a else+while for the multi-line file